pkg/audit/disqualification_test.go

```diff
@@ -196,6 +196,45 @@ func TestDisqualifiedNodesGetNoUpload(t *testing.T) {
 	})
 }
 
+func TestDisqualifiedNodesRemainsDisqualified(t *testing.T) {
+
+	// - mark a node as disqualified
+	// - give it high uptime and audit rate
+	// - check that the node remains disqualified
+
+	testplanet.Run(t, testplanet.Config{
+		SatelliteCount: 1, StorageNodeCount: 4, UplinkCount: 1,
+	}, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
+		satellite := planet.Satellites[0]
+
+		err := satellite.Audit.Service.Close()
+		require.NoError(t, err)
+
+		disqualifiedNode := planet.StorageNodes[0]
+		disqualifyNode(t, ctx, satellite, disqualifiedNode.ID())
+
+		_, err = satellite.DB.OverlayCache().UpdateUptime(ctx, disqualifiedNode.ID(), true, 1, 1, 1)
+		require.NoError(t, err)
+
+		assert.True(t, isDisqualified(t, ctx, satellite, disqualifiedNode.ID()))
+
+		_, err = satellite.DB.OverlayCache().UpdateStats(ctx, &overlay.UpdateRequest{
+			NodeID:       disqualifiedNode.ID(),
+			IsUp:         true,
+			AuditSuccess: true,
+			AuditLambda:  1,
+			AuditWeight:  1,
+			AuditDQ:      1,
```

Comment:
Set AuditDQ/UptimeDQ to 0 to guarantee that the new reputation scores are larger than the DQ thresholds. That way, the test is verifying that even though reputation is undoubtedly larger than the dq threshold, the node remains disqualified